issue-scan

```diff
@@ -134,31 +134,43 @@ def output_task(command, issue, repo, verbose):
     if "pull_request" in issue or "commits" in issue:
         ref = f"pull/{number}/head"
         sha = api.get(f"git/ref/{ref}")["object"]["sha"]
+        pull = number
     else:
         ref = testmap.get_default_branch(repo)
         sha = api.get(f"git/ref/heads/{ref}")["object"]["sha"]
+        pull = None
 
     if verbose:
         return f"issue-{number} {name} {context} {ref}"
 
     current = time.strftime('%Y%m%d-%H%M%S')
+    slug = f'{name}-{context}-{sha[:8]}-{current}'
+
     cmd = f"./s3-streamer --repo {repo} --test-name {name}-{context}-{current} " \
         f"--github-context {name}/{context} --revision {sha} -- sh -exc '"
     cmd += f"./make-checkout --verbose --repo {repo} {ref}; cd make-checkout-workdir; "
     cmd += f"./{name} --verbose --issue={number} {context}'"
 
-    return cmd
-
-
-def queue_task(channel, result):
-    body = {
-        "command": result,
+    return {
+        'job': {
+            'repo': repo,
+            'sha': sha,
+            'pull': pull,
+            'slug': slug,
+            'context': f'{name}/{context}',
+            'command': (f'./{name}', '--verbose', f'--issue={number}', context),
+            'secrets': ('github-token', 'image-upload'),
+        },
+        "command": cmd,
         "type": "issue",
     }
-    queue = 'rhel' if contains_any(result, REDHAT_TASKS) else 'public'
+
+
+def queue_task(channel, body):
+    queue = 'rhel' if contains_any(channel['job']['context'], REDHAT_TASKS) else 'public'
     channel.basic_publish('', queue, json.dumps(body),
                           properties=pika.BasicProperties(priority=distributed_queue.MAX_PRIORITY))
-    logging.info("Published issue task with command: '%s'", result)
+    logging.info("Published issue task with command: '%s'", channel['command'])
```

Comment:
`channel['command']` looks like "I did this at 23:00" :grin: 

How about just printing the entire `body`, and `s/ with command//` ? Useful for debugging.

Reply:
#6026 spots this, and also brings the scaffolding for validating the `job` JSON. So let's merge that first, rebase this and update the tests?